src/sql/util.py

```diff
@@ -157,6 +157,99 @@ def show_deprecation_warning():
     )
 
 
+def check_duplicate_arguments(cmd_from, args) -> bool:
+    """
+    Raises UsageError when duplicate arguments are passed to magics.
+    Returns true if no duplicates in arguments or aliases.
+    """
+    if len(args) <= 1:
+        return True
+
+    allowed_duplicates = {
+        "sql": ["-w", "--with", "--append", "--interact"],
+        "sqlplot": ["-w", "--with"],
+        "sqlcmd": [],
+    }
+
+    disallowed_aliases = {
+        "sql": {
+            "-l": "--connections",
+            "-x": "--close",
+            "-c": "--creator",
+            "-s": "--section",
+            "-p": "--persist",
+            "-a": "--connection-arguments",
+            "-f": "--file",
+            "-n": "--no-index",
+            "-S": "--save",
+            "-A": "--alias",
+        },
+        "sqlplot": {
+            "-t": "--table",
+            "-s": "--schema",
+            "-c": "--column",
+            "-o": "--orient",
+            "-b": "--bins",
+            "-B": "--breaks",
+            "-W": "--binwidth",
+            "-S": "--show-numbers",
+        },
+        "sqlcmd": {
+            "-t": "--table",
+            "-s": "--schema",
+            "-o": "--output",
+        },
+    }
+
+    args = [arg for arg in args if arg.startswith("--") or arg.startswith("-")]
+
+    single_hyphen_opts = set()
+    double_hyphen_opts = set()
+
+    for arg in args:
+        if arg.startswith("--"):
+            double_hyphen_opts.add(arg)
+        elif arg.startswith("-"):
+            single_hyphen_opts.add(arg)
+
+    duplicate_args = []
+    visited_args = set()
+    for arg in args:
+        if arg not in allowed_duplicates[cmd_from]:
+            if arg not in visited_args:
+                visited_args.add(arg)
+            else:
+                duplicate_args.append(arg)
+
+    either_or = [
+        (opt, disallowed_aliases[cmd_from][opt])
+        for opt in single_hyphen_opts
+        if opt in disallowed_aliases[cmd_from]
+        if disallowed_aliases[cmd_from][opt] in double_hyphen_opts
+    ]
+
+    error_message = ""
+    if duplicate_args:
+        error_message += (
```

Comment:
please use string format here instead of `+`

Reply:
Fixed [here](https://github.com/ploomber/jupysql/pull/908/files#diff-653c15e6161c97f356be04e072838ed15014af275c06019a7692ab727fa44b97R231-R251)